contracts/DAORegistryScheme.sol

```diff
@@ -0,0 +1,192 @@
+        parameters[paramsHash].contractToCall = _contractToCall;
+        return paramsHash;
+    }
+
+    function getParametersHash(
+        bytes32 _voteParams,
+        IntVoteInterface _intVote,
+        address _contractToCall
+    ) public pure returns(bytes32)
+    {
+        return keccak256(abi.encodePacked(_voteParams, _intVote, _contractToCall));
+    }
+
+    /**
+    * @dev create a proposal to register a DAO with to a named registry
+    * @param _avatar the address of the organization the resource will be registered for
+    * @param _registryName the name of the registry to add the resource
+    * @param _proposedAvatar the organization we want to add to the registry
+    * @param _callData the abi call to add _proposedAvatar to the registry
+    * @param _value value(ETH) to transfer with the call
+    * @return a proposal Id
+    */
+    function proposeToAddDAO(
+        Avatar _avatar,
+        string memory _registryName,
+        Avatar _proposedAvatar,
+        bytes memory _callData,
+        uint256 _value
+    ) public returns(bytes32)
+    {
+        // propose
+        Parameters memory controllerParams = parameters[getParametersFromController(_avatar)];
+
+        bytes32 proposalId = controllerParams.intVote.propose(
+            2,
+            controllerParams.voteParams,
+            msg.sender,
+            address(_avatar)
+        );
+
+        DAORegistryProposal memory proposal = DAORegistryProposal({
+            avatar: address(_proposedAvatar),
+            registryName: _registryName,
+            addDAO: true,
+            callData: _callData,
+            value: _value
+        });
+
+        emit NewDAOProposal(
+            address(_avatar),
+            proposalId,
+            _registryName,
+            address(_proposedAvatar)
+        );
+
+        organizationsProposals[address(_avatar)][proposalId] = proposal;
+        proposalsInfo[address(controllerParams.intVote)][proposalId] = ProposalInfo({
+            blockNumber:block.number,
+            avatar:_avatar
+        });
+
+        return proposalId;
+    }
+
+    /**
+    * @dev propose to remove a DAO inside a named registry
+    * @param _avatar the address of the controller from which we want to remove a scheme
+    * @param _registryName the name of the registry we want to remove from
+    * @param _proposedAvatar the organization we want to remove from the registry
+    * @param _callData the abi encoded call to remove the registry
+    * @param _value value(ETH) to transfer with the call
+    */
+    function proposeToRemoveDAO(
+        Avatar _avatar,
+        string memory _registryName,
+        Avatar _proposedAvatar,
+        bytes memory _callData,
+        uint256 _value
+    ) public returns(bytes32)
+    {
+        bytes32 paramsHash = getParametersFromController(_avatar);
+        Parameters memory params = parameters[paramsHash];
+
+        IntVoteInterface intVote = params.intVote;
```

Comment:
this part is duplicated on the proposetoadd dao. 
how about merge common code...?